sdk/scheduler/src/main/java/com/mesosphere/sdk/specification/DefaultTransportEncryptionSpec.java

```diff
@@ -2,39 +2,33 @@
 
 import com.fasterxml.jackson.annotation.JsonCreator;
 import com.fasterxml.jackson.annotation.JsonProperty;
-import com.mesosphere.sdk.specification.validation.ValidationUtils;
 import org.apache.commons.lang3.builder.EqualsBuilder;
 import org.apache.commons.lang3.builder.HashCodeBuilder;
 import org.apache.commons.lang3.builder.ReflectionToStringBuilder;
 
-import javax.validation.Valid;
-import javax.validation.constraints.NotNull;
-import javax.validation.constraints.Size;
-
 /**
  * Default implementation of {@link TransportEncryptionSpec}.
  */
 public class DefaultTransportEncryptionSpec implements TransportEncryptionSpec {
 
-    @Valid
-    @NotNull(message = "Transport encryption name cannot be empty")
-    @Size(min = 1, message = "Transport encryption name cannot be empty")
-    private String name;
-
-    @Valid
-    private Type type;
+    private final String name;
+    private final Type type;
 
     @JsonCreator
-    public DefaultTransportEncryptionSpec(
+    private DefaultTransportEncryptionSpec(
             @JsonProperty("name") String name,
             @JsonProperty("type") Type type) {
         this.name = name;
         this.type = type;
     }
 
+    public static Builder newBuilder() {
+        return new Builder();
+    }
+
     public DefaultTransportEncryptionSpec(Builder builder) {
         this(builder.name, builder.type);
-        ValidationUtils.validate(this);
+        ValidationUtils.nonEmpty(this, "name", name);
```

Comment:
Just thinking about this, would a `nonBlank` check make sense too?

Reply:
Yes. Done. Also used it for PodSpec.type.